Add ignore lists

Some files are invalid on purpose (for testing). How to deal with them?

```
lando grum --tasks yaml_lint
GrumPHP is sniffing your code!
Running task 1/1: YamlLintTask... ✘
[ERROR] Indentation problem in "./web/core/tests/Drupal/Tests/Core/Asset/library_test_files/invalid_file.libraries.yml" at line 3 (near "  key2: value2").
```

--- a/README.md
+++ b/README.md
@@ -5,8 +5,8 @@
 This composer package will provide some basic code quality checks before committing
 code by using https://github.com/phpro/grumphp.
 
-It checks only modified files or new files on git commit, but check on all configured
-paths can be executed running `vendor/bin/grumphp run`
+**It checks only modified files or new files on git commit, but check on all configured
+paths can be executed running `vendor/bin/grumphp run`**
 
 This tool only extends [GrumPHP](https://github.com/phpro/grumphp) please read 
 its [documentation](https://github.com/phpro/grumphp/blob/master/README.md#configuration) on how to configure tool itself.

--- a/README.md
+++ b/README.md
@@ -65,7 +65,7 @@ parameters:
 
 Same applies to any task that uses other configuration file (easy-coding-standards).
 
-Ech code quality tool allows you to define at least 3 things:
+**Each code quality tool allows you to define at least 3 things:**
 - `run_on` - Multiple paths that will be checked and files staged must be from same path
 - `ignore_patterns` - path parts that will exclude files from check
 - `extensions` - file extensions of files that should be checked

--- a/README.md
+++ b/README.md
@@ -89,12 +89,12 @@ Since GrumPHP is just a CLI tool, these commands can be triggered:
 
 The pre-commit hook will be automatically run upon executing `git commit`.
 
-The code scanning can be avoided by `git commit --no-verify`.
+The code scanning can be avoided by `git commit --no-verify` or `git commit -n`.
 
-You can run the checks manually with: `./bin/grumphp run`
+You can run the checks manually with: `./vendor/bin/grumphp run`
 
 ## Usage in Continuous Integration
-You can easily use the code quality checkers on your CI (Jenkins/GitLab CI) by adding this line:
+You can easily use the code quality checkers on your CI (CircleCi/Jenkins/GitLab CI) by adding this line:
 
 ```
 ./vendor/bin/grumphp run --no-ansi --no-interaction

--- a/README.md
+++ b/README.md
@@ -2,27 +2,31 @@
 
 [![CircleCI](https://circleci.com/gh/wunderio/code-quality.svg?style=svg)](https://circleci.com/gh/wunderio/code-quality)
 
-This composer package will provide some basic code quality checks before commiting code by using
-https://github.com/phpro/grumphp.
+This composer package will provide some basic code quality checks before committing
+code by using https://github.com/phpro/grumphp.
 
-It checks only modified files or new files.
+It checks only modified files or new files on git commit, but check on all configured
+paths can be executed running `vendor/bin/grumphp run`
+
+This tool only extends [GrumPHP](https://github.com/phpro/grumphp) please read 
+its [documentation](https://github.com/phpro/grumphp/blob/master/README.md#configuration) on how to configure tool itself.
 
 ## Checks performed
 
 This repository currently has following checks:
 
-* PHP Drupal Coding Standards
-* PHP 7.3 Compatibility
-* PHP syntax
-* Shell script exec bits
-* PHP Code security
-* Cognitive complexity
-* Yaml syntax
-* Json syntax
+* Shell script exec bits - [check_file_permissions](src/Task/CheckFilePermissions/README.md)
+* PHP Drupal CS and PHP Code security  - [phpcs](src/Task/Phpcs/README.md)
+* PHP 7.3 Compatibility - [php_compatibility](src/Task/PhpCompatibility/README.md)
+* PHP syntax - [php_check_syntax](src/Task/PhpCheckSyntax/README.md)
+* Cognitive complexity and other ecs sniffs - [ecs](src/Task/Ecs/README.md)
+* Yaml syntax - [yaml_lint](src/Task/YamlLint/README.md)
+* Json syntax - [json_lint](src/Task/JsonLint/README.md)
 
 ## Pre-requisites
 
 * Composer
+* PHP >= 7.1
 
 ## Installation
 
@@ -35,8 +39,20 @@ cp vendor/wunderio/code-quality/config/grumphp.yml ./grumphp.yml
 
 The commit hook for GrumPHP is automatically installed on composer require.
 
-## Custom PHP CodeSniffer rules
+## Customization
 
+### Configuration
+
+Details of the configuration are broken down into the following sections.
+
+- [Parameters](https://github.com/phpro/grumphp/blob/master/doc/parameters.md) &ndash; Configuration settings for GrumPHP itself.
+- [Tasks](https://github.com/phpro/grumphp/blob/master/doc/tasks.md) &ndash; External tasks performing code validation and their respective configurations.
+- [TestSuites](https://github.com/phpro/grumphp/blob/master/doc/testsuites.md)
+- [Extensions](https://github.com/phpro/grumphp/blob/master/doc/extensions.md)
+- [Events](https://github.com/phpro/grumphp/blob/master/doc/events.md)
+- [Conventions checker](https://github.com/phpro/grumphp/blob/master/doc/conventions.md)
+
+### Task parameters
 If you need to customize the rules for PHP CodeSniffer then drop in phpcs.xml in the same
 folder as composer.json and configure grumphp.yml:
 ````yml
@@ -46,6 +62,28 @@ parameters:
       standard:
         - phpcs.xml
 ````
+
+Same applies to any task that uses other configuration file (easy-coding-standards).
+
+Ech code quality tool allows you to define at least 3 things:
+- `run_on` - Multiple paths that will be checked and files staged must be from same path
+- `ignore_patterns` - path parts that will exclude files from check
+- `extensions` - file extensions of files that should be checked
+
+Please see individual task documentation for more information on what are other configurable options.
+
+
+
+## Commands
+
+Since GrumPHP is just a CLI tool, these commands can be triggered:
+
+- [configure](https://github.com/phpro/grumphp/blob/master/doc/commands.md#installation)
+- [git:init](https://github.com/phpro/grumphp/blob/master/doc/commands.md#installation)
+- [git:deinit](https://github.com/phpro/grumphp/blob/master/doc/commands.md#installation)
+- [git:pre-commit](https://github.com/phpro/grumphp/blob/master/doc/commands.md#git-hooks)
+- [git:commit-msg](https://github.com/phpro/grumphp/blob/master/doc/commands.md#git-hooks)
+- [run](https://github.com/phpro/grumphp/blob/master/doc/commands.md#run)
 
 ## Usage
 
@@ -59,5 +97,5 @@ You can run the checks manually with: `./bin/grumphp run`
 You can easily use the code quality checkers on your CI (Jenkins/GitLab CI) by adding this line:
 
 ```
-./bin/grumphp run --no-ansi --no-interaction
+./vendor/bin/grumphp run --no-ansi --no-interaction
 ```

--- a/src/Task/CheckFilePermissions/README.md
+++ b/src/Task/CheckFilePermissions/README.md
@@ -2,14 +2,21 @@
 
 Check if the file permissions match that you've required.
 
-### grumphp.yml:
+
+### grumphp.yml (with current defaults):
 ````yml
 parameters:
     tasks:
         check_file_permissions:
-            run_on: ['.']
-            extensions: [php, inc, module, phtml, php3, php4, php5]
-            ignore_patterns: ['*/vendor/*','*/node_modules/*']
+            ignore_patterns:
+                - '/vendor/'
+                - '/node_modules/'
+                - '/core/'
+                - '/libraries/'
+            extensions:
+              defaults: ['sh']
+            run_on:
+              defaults: ['.']
     extensions:
         - Wunderio\GrumPHP\Task\CheckFilePermissions\CheckFilePermissionsExtensionLoader
 ````

--- a/src/Task/Ecs/README.md
+++ b/src/Task/Ecs/README.md
@@ -2,17 +2,22 @@
 
 Easy coding standards implementation to better react on task context.
 
-### grumphp.yml:
+### grumphp.yml (with current defaults):
 ````yml
 parameters:
     tasks:
         ecs:
+            ignore_patterns:
+                - '/vendor/'
+                - '/node_modules/'
+                - '/core/'
+                - '/libraries/'
+            extensions: ['php', 'inc', 'module', 'install']
             run_on: ['.']
-            extensions: [php, inc, module, phtml, php3, php4, php5]
-            ignore_patterns: ['*/vendor/*','*/node_modules/*']
-            config: ecs.yml
             clear-cache: false
+            config: 'vendor/wunderio/code-quality/config/ecs.yml'
             no-progress-bar: true
+            level: ~
     extensions:
         - Wunderio\GrumPHP\Task\Ecs\EcsExtensionLoader
 ````

--- a/src/Task/JsonLint/README.md
+++ b/src/Task/JsonLint/README.md
@@ -2,14 +2,18 @@
 
 Json linter.
 
-### grumphp.yml:
+### grumphp.yml (with current defaults):
 ````yml
 parameters:
     tasks:
         json_lint:
+            ignore_patterns:
+                - '/vendor/'
+                - '/node_modules/'
+                - '/core/'
+                - '/libraries/'
+            extensions: ['json', 'lock']
             run_on: ['.']
-            extensions: ['yaml', 'yml']
-            ignore_patterns: ['*/vendor/*','*/node_modules/*']
             detect_key_conflicts: false
     extensions:
         - Wunderio\GrumPHP\Task\JsonLint\JsonLintExtensionLoader

--- a/src/Task/PhpCheckSyntax/README.md
+++ b/src/Task/PhpCheckSyntax/README.md
@@ -2,14 +2,19 @@
 
 Run `php -l` on the code.
 
-### grumphp.yml:
+### grumphp.yml (with current defaults):
 ````yml
 parameters:
     tasks:
         php_check_syntax:
-            run_on: ['.']
-            extensions: [php, inc, module, phtml, php3, php4, php5]
-            ignore_patterns: ['*/vendor/*','*/node_modules/*']
+            ignore_patterns:
+                - '/vendor/'
+                - '/node_modules/'
+                - '/core/'
+                - '/libraries/'
+            extensions: ['php', 'inc', 'module', 'install']
+            run_on: '.']
+            parallelism: 100
     extensions:
         - Wunderio\GrumPHP\Task\PhpCheckSyntaxTask\PhpCheckSyntaxExtensionLoader
 ````

--- a/src/Task/PhpCheckSyntax/README.md
+++ b/src/Task/PhpCheckSyntax/README.md
@@ -13,7 +13,7 @@ parameters:
                 - '/core/'
                 - '/libraries/'
             extensions: ['php', 'inc', 'module', 'install']
-            run_on: '.']
+            run_on: ['.']
             parallelism: 100
     extensions:
         - Wunderio\GrumPHP\Task\PhpCheckSyntaxTask\PhpCheckSyntaxExtensionLoader

--- a/src/Task/PhpCompatibility/README.md
+++ b/src/Task/PhpCompatibility/README.md
@@ -2,16 +2,21 @@
 
 Check if files are compatible with X version of PHP.
 
-### grumphp.yml:
+### grumphp.yml (with current defaults):
 ````yml
 parameters:
     tasks:
         php_compatibility:
-            testVersion: "7.3"
-            standard: "php-compatibility.xml"
+            ignore_patterns: 
+                - '/vendor/'
+                - '/node_modules/'
+                - '/core/'
+                - '/libraries/'
+            extensions: ['php', 'inc', 'module', 'install']
             run_on: ['.']
-            extensions: [php, inc, module, phtml, php3, php4, php5]
-            ignore_patterns: ['*/vendor/*','*/node_modules/*']
+            testVersion: '7.3'
+            standard: 'vendor/wunderio/code-quality/config/php-compatibility.xml'
+            parallel: 20
     extensions:
         - Wunderio\GrumPHP\Task\PhpCompatibilityTask\PhpCompatibilityExtensionLoader
 ````

--- a/src/Task/Phpcs/README.md
+++ b/src/Task/Phpcs/README.md
@@ -1,17 +1,32 @@
 # phpcs
 
-Check Drupal code against coding standards and other best practices.
+Check Drupal code against coding standards and security standards.
 
-### grumphp.yml:
+### grumphp.yml (with current defaults):
 ````yml
 parameters:
     tasks:
         phpcs:
-            standard:
-                - phpcs.xml
+            ignore_patterns: 
+                - '/vendor/'
+                - '/node_modules/'
+                - '/core/'
+                - '/libraries/'
+            extensions: ['php', 'inc', 'module', 'install']
             run_on: ['.']
-            extensions: [php, inc, module, phtml, php3, php4, php5]
-            ignore_patterns: ['*/vendor/*','*/node_modules/*']
+            standard:
+                - 'vendor/wunderio/code-quality/config/phpcs.xml'
+                - 'vendor/wunderio/code-quality/config/phpcs-security.xml'
+            tab_width: ~
+            encoding: ~
+            sniffs: []
+            severity: ~
+            error_severity:  ~
+            warning_severity: ~
+            report: 'full'
+            report_width: 120
+            exclude: []
+            parallel: 20
     extensions:
         - Wunderio\GrumPHP\Task\PhpCompatibilityTask\PhpCompatibilityExtensionLoader
 ````

--- a/src/Task/YamlLint/README.md
+++ b/src/Task/YamlLint/README.md
@@ -2,18 +2,22 @@
 
 YAML linter.
 
-### grumphp.yml:
+### grumphp.yml (with current defaults):
 ````yml
 parameters:
     tasks:
         yaml_lint:
-            run_on: ['.']
+            ignore_patterns:
+                - '/vendor/'
+                - '/node_modules/'
+                - '/core/'
+                - '/libraries/'
             extensions: ['yaml', 'yml']
-            ignore_patterns: ['*/vendor/*','*/node_modules/*']
+            run_on: ['.']
             object_support: false
             exception_on_invalid_type: false
-            parse_constant:  false
-            parse_custom_tags: false
+            parse_constant: false
+            parse_custom_tags: true
     extensions:
         - Wunderio\GrumPHP\Task\YamlLint\YamlLintExtensionLoader
 ````

--- a/src/Task/tasks.yml
+++ b/src/Task/tasks.yml
@@ -171,6 +171,8 @@ Wunderio\GrumPHP\Task\YamlLint\YamlLintTask:
       defaults:
         - '/vendor/'
         - '/node_modules/'
+        - '/core/'
+        - '/libraries/'
       allowed_types: ['array']
     extensions:
       defaults: ['yaml', 'yml']
@@ -200,6 +202,8 @@ Wunderio\GrumPHP\Task\JsonLint\JsonLintTask:
       defaults:
         - '/vendor/'
         - '/node_modules/'
+        - '/core/'
+        - '/libraries/'
       allowed_types: ['array']
     extensions:
       defaults: ['json', 'lock']

